src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/TreatIterator.java

```diff
@@ -15,40 +16,48 @@
 import sparksoniq.semantics.types.SequenceType;
 
 import java.util.Collections;
-import java.util.List;
 
 
 public class TreatIterator extends HybridRuntimeIterator {
 
     private static final long serialVersionUID = 1L;
     private RuntimeIterator _iterator;
+    private SequenceType _sequenceType;
+    private boolean _shouldThrowTreatException;
+
+    private ItemType itemType;
+    private String sequenceTypeName;
+
     private Item _nextResult;
-    private final SequenceType _sequenceType;
     private int _childIndex;
-    private final ItemType itemType;
-    private final String sequenceTypeName;
 
-    public TreatIterator(RuntimeIterator iterator, SequenceType sequenceType, IteratorMetadata iteratorMetadata) {
+    public TreatIterator(
+            RuntimeIterator iterator,
+            SequenceType sequenceType,
+            boolean shouldThrowTreatException,
```

Comment:
Is this boolean still needed, now that function type checking is handled with an iterator of its own?

Reply:
The boolean is not needed for this branch, but I will push the TreatIterator with the boolean in #379, so I guess I will just leave it here for faster merge/conflict resolution.

Reply:
Yes, this works. Thank you for the precision.